### PR TITLE
Add unit tests for _get_type_id

### PR DESCRIPTION
**Pull request by maksimdrachov**
## Summary by CodeRabbit

- **Tests**
	- Added new assertions to existing test cases to verify the expected type ID for the `repeat()`, `poll()`, and `defer()` methods in the event loop testing suite.

### CODE REVIEW

File: tests/test_olg_scheduler.cpp [hunk 1]
```diff
@@ -85,6 +85,12 @@ TEST(TestOlgScheduler, EventLoopBasic)
     using Arg  = Arg<Loop::time_point>;
     Loop evl;
 
+    // Check the type ID.
+    constexpr std::array<std::uint8_t, 16> expected_type_id =
+        {0xB6, 0x87, 0x48, 0xA6, 0x7A, 0xDB, 0x4D, 0xF1, 0xB3, 0x1D, 0xA9, 0x8D, 0x50, 0xA7, 0x82, 0x47};
+    const auto actual_type_id = Event<Loop::time_point>::_get_type_id_();
+    EXPECT_THAT(actual_type_id, testing::ElementsAreArray(expected_type_id));
```
[the remark this answers is not in the document]
Comment (serges147) [reply tests/test_olg_scheduler.cpp]: I agree with rabbit about separate dedicated test. Also, you test with `Event<>` but the main point was that result (Voldemort) types of `repeat`, `defer` and `poll` expose the type id, so I would rather test them than internal `Event`.

[the remark this answers is not in the document]
Comment (maksimdrachov) [reply tests/test_olg_scheduler.cpp]: Fixed.

**Review by serges147 — APPROVED** [review]
`ElementsAreArray` is needed if C arrays are compared. For c++ collections you can use `ElementsAre` (there is already `using testing::ElementsAre;` BTW).

**Comment by maksimdrachov** [comment]
It seems to me like it's cleaner to use `ElementsAreArray`? (That way I only have to define the type ID only once)

image

If I replace with `ElementsAre`, I get the following compilation error:

image

**Pull request merged**